apis/src/components/organisms/quickplay.rs

```diff
@@ -0,0 +1,83 @@
+use core::panic;
+use leptos::*;
+
+use leptos_router::use_navigate;
+
+use crate::{
+    common::ChallengeAction,
+    providers::{ApiRequests, AuthContext},
+};
+use hive_lib::{ColorChoice, GameType};
+use shared_types::ChallengeVisibility;
+use shared_types::{ChallengeDetails, TimeMode};
+
+#[component]
+fn GridButton(text: String) -> impl IntoView {
+    let display_text = text.clone();
+    view! {
+        <button
+            class="px-4 py-2 font-bold text-white rounded bg-button-dawn dark:bg-button-twilight hover:bg-pillbug-teal active:scale-95"
+            on:click=move |_| {
+                let api = ApiRequests::new();
+                let auth_context = expect_context::<AuthContext>();
+                let account = match (auth_context.user)() {
+                    Some(Ok(Some(account))) => Some(account),
+                    _ => None,
+                };
+                let (base, increment) = match text.as_str() {
+                    "1+0" => (1, 0),
+                    "2+1" => (2, 1),
+                    "3+0" => (3, 0),
+                    "3+2" => (3, 2),
+                    "5+0" => (5, 0),
+                    "5+3" => (5, 3),
+                    "10+0" => (10, 0),
+                    "10+5" => (10, 5),
+                    "15+10" => (15, 10),
+                    _ => panic!("Invalid text"),
+                };
+                if account.is_some() {
+                    let details = ChallengeDetails {
+                        rated: false,
+                        game_type: GameType::MLP,
+                        visibility: ChallengeVisibility::Public,
+                        opponent: None,
+                        color_choice: ColorChoice::Random,
+                        time_mode: TimeMode::RealTime,
+                        time_base: Some(base * 60),
+                        time_increment: Some(increment),
+                        band_upper: None,
+                        band_lower: None,
+                    };
+                    let challenge_action = ChallengeAction::Create(details);
+                    api.challenge(challenge_action);
+                } else {
+                    let navigate = use_navigate();
+                    navigate("/login", Default::default());
+                }
+            }
+        >
+
+            {display_text}
+        </button>
+    }
+}
+#[component]
+pub fn QuickPlay() -> impl IntoView {
+    view! {
+        <div>
+            <span class="text-xl font-bold">Quick Play</span>
+            <div class="flex grid grid-cols-2 gap-4 w-full md:grid-cols-3">
```

Comment:
flex and grid apply the same property, and it can lead to weird behavior between browsers because they apply the classes slightly differently. Just grid here